### PR TITLE
[NON-MODULAR] [READY] More people who can antag who shouldn't be able to

### LINKED ISSUE #6406: NT rep can roll mid round antag, heretic.
Reporting client version: 514.1547

## Round ID:
2126

## Testmerges:
- [S&T Additional Emotes](https://github.com/Skyrat-SS13/Skyrat-tg/pull/6259)
- [Synthetic wounds](https://github.com/Skyrat-SS13/Skyrat-tg/pull/6369)

### PR DESCRIPTION
**Pull request by lyricalpaws**
## About The Pull Request
(Hopefully) fixes #6406 and a few other edge cases 

## Why It's Good For The Game
These roles shouldn't be able to roll antag, and this makes sure that they can't.

## Changelog
:cl:
fix: A lot of security, BS, and NTR can no longer roll certain antags as referenced in #6406 
server: changes to `config/dynamic.json` should be reflected
/:cl:

### CODE REVIEW

**Comment by Vizoplez** [comment]
👍 

looks good.

**Comment by unit0016** [comment]
I still have a giggle every time a ghostroles manages to roll antag, assumedly thanks to the create antag verb.
Yee honk.

**Comment by hugmeorelse** [comment]
This is a config change. I'll manually update it on the box. I will ask for this to be closed once that is completed tomorrow.

**Comment by lyricalpaws** [comment]
There's changes to files too, not just to configs

**Comment by KathrinBailey** [comment]
> This is a config change. I'll manually update it on the box. I will ask for this to be closed once that is completed tomorrow.

Urrhh? @hugmeorelse update?

**Comment by lyricalpaws** [comment]
> 
> 
> > This is a config change. I'll manually update it on the box. I will ask for this to be closed once that is completed tomorrow.
> 
> Urrhh? @hugmeorelse update?

hugmeorelse's comment is moot. There's changes to files outside of config too.

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm [hunk 1]
```diff
@@ -73,7 +73,7 @@
 	antag_datum = /datum/antagonist/rev/head
 	antag_flag = ROLE_REV_HEAD
 	antag_flag_override = ROLE_REV
-	restricted_roles = list("AI", "Cyborg", "Prisoner", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Chief Engineer", "Corrections Officer", "Chief Medical Officer", "Research Director", "Vanguard Operative")
+	restricted_roles = list("AI", "Cyborg", "Prisoner", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Chief Engineer", "Corrections Officer", "Chief Medical Officer", "Research Director", "Vanguard Operative", "Nanotrasen Representative", "Blueshield", "Civil Disputes Officer") //SKYRAT EDIT - Sec_haul
```
Comment (tf-4) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm]:
```suggestion
	restricted_roles = list("AI", "Cyborg", "Prisoner", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Chief Engineer", "Corrections Officer", "Chief Medical Officer", "Research Director", "Vanguard Operative", "Nanotrasen Representative", "Blueshield", "Civil Disputes Officer", "Security Sergeant") //SKYRAT EDIT - Sec_haul
```

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 2]
```diff
@@ -250,7 +250,7 @@
 	persistent = TRUE
 	antag_flag = ROLE_FAMILIES
 	protected_roles = list("Prisoner", "Head of Personnel")
-	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Corrections Officer", "Vanguard Operative")
+	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Corrections Officer", "Vanguard Operative", "Nanotrasen Representative", "Blueshield", "Civil Disputes Officer") //SKYRAT EDIT - Sec_haul
```
Comment (tf-4) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm]:
```suggestion
	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Corrections Officer", "Vanguard Operative", "Nanotrasen Representative", "Blueshield", "Civil Disputes Officer", "Security Sergeant") //SKYRAT EDIT - Sec_haul
```

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 3]
```diff
@@ -496,7 +496,7 @@
 	persistent = TRUE
 	antag_flag = ROLE_FAMILIES
 	protected_roles = list("Prisoner", "Head of Personnel")
-	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Corrections Officer", "Vanguard Operative")
+	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Corrections Officer", "Vanguard Operative", "Nanotrasen Representative", "Blueshield", "Civil Disputes Officer")
```
Comment (tf-4) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]:
```suggestion
	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Corrections Officer", "Vanguard Operative", "Nanotrasen Representative", "Blueshield", "Civil Disputes Officer", "Security Sergeant") //SKYRAT EDIT - SEC_HAUL
```

**Review by tf-4 — COMMENTED** [review]
Can you update `modular_skyrat/modules/oneclickantag/oneclickantag.dm` with NT Rep as well?

File: config/dynamic.json [hunk 4]
```diff
@@ -24,11 +24,22 @@
 			},
 			"protected_roles": [
```
Comment (JohnFulpWillard) [inline config/dynamic.json:25]: This and other antags are all done through config, is the TG edits worth editing things that should be done in config instead?

**Comment by hugmeorelse** [comment]
bump

**Review by Funce — APPROVED** [review]
This is fine. Repo config should represent as close to live as possible so testers can test with the most accurate settings.

**Pull request merged**